libs/providers/flipt-web/src/lib/flipt-web-provider.ts

```
@@ -0,0 +1,263 @@
+          };
+        case EvaluationReason.UNKNOWN:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.UNKNOWN,
+          };
+        default:
+          return {
+            value: defaultValue,
+            reason: StandardResolutionReasons.DEFAULT,
+          };
+      }
+    } catch (e) {
+      throw new GeneralError(getErrorMessage(e));
+    }
+  }
+
+  resolveStringEvaluation(
+    flagKey: string,
+    defaultValue: string,
+    context: EvaluationContext,
+  ): ResolutionDetails<string> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'string', defaultValue, mergedContext);
+    return value as ResolutionDetails<string>;
+  }
+
+  resolveNumberEvaluation(
+    flagKey: string,
+    defaultValue: number,
+    context: EvaluationContext,
+  ): ResolutionDetails<number> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'number', defaultValue, mergedContext);
+    return value as ResolutionDetails<number>;
+  }
+
+  resolveObjectEvaluation<U extends JsonValue>(
+    flagKey: string,
+    defaultValue: U,
+    context: EvaluationContext,
+  ): ResolutionDetails<U> {
+    const mergedContext = { ...this._globalContext, ...context };
+    const value = this.resolveFlagHelper(flagKey, 'json', defaultValue, mergedContext);
+    return value as ResolutionDetails<U>;
+  }
+
+  private resolveFlagHelper<U extends JsonValue>(
+    flagKey: string,
+    flagType: PrimitiveTypeName,
+    defaultValue: PrimitiveType | U,
+    context: EvaluationContext,
+  ): ResolutionDetails<PrimitiveType | U> {
+    const evalContext: Record<string, string> = transformContext(context);
+
+    try {
+      const resp = this._client?.evaluateVariant(flagKey, context.targetingKey ?? '', evalContext);
+
+      if (resp?.status === 'failure') {
+        return {
```

You can throw the general error with a message. The SDK will handle the response for you.